workspaces/ui-v2/src/components/sharing/ShareModal.tsx

```diff
@@ -0,0 +1,223 @@
+  };
+
+  const queryParams = Object.entries(params)
+    .map(([k, v]) => `${k}=${encodeURIComponent(v)}`)
+    .join('&');
+
+  return `https://img.shields.io/static/v1?${queryParams}`;
+}
+
+function generateBadgeMarkdown({
+  endpointCount,
+  shareUrl,
+}: {
+  endpointCount: number;
+  shareUrl: string;
+}) {
+  return `[![API Docs: ${endpointCount} Endpoints](${generateBadgeUrl({
+    endpointCount,
+  })})](${shareUrl})`;
+}
+
+export const ShareModal: React.FC<{
+  share: (intent: ShareTarget) => Promise<string>;
+}> = (props) => {
+  const styles = useStyles(props);
+
+  const { isAuthenticated, isLoading, loginWithPopup } = useAuth0();
+
+  const endpointCount = useAppSelector(
+    (state) => state.endpoints.results?.data!.length || 0
+  );
+
+  const [shareType, setShareType] = useState<ShareTarget | null>(null);
+
+  let inputRef = useRef<HTMLInputElement>();
+  const [copied, setCopied] = useState(false);
+
+  useEffect(() => {
+    if (copied) {
+      let timeout = setTimeout(() => setCopied(false), 7500);
+      return () => clearTimeout(timeout);
+    }
+  }, [copied, setCopied]);
+
+  const copy = useCallback(() => {
+    const input = inputRef.current;
+    if (input) {
+      input.focus();
+      input.select();
+      document.execCommand('copy');
+      setCopied(true);
+    }
+  }, [inputRef, setCopied]);
+
+  // Call share when authenticated and you set a shareType
+  const shareUrl = useAsyncMemo(async () => {
```

Comment:
Sort of minor, but since this is an async state - we should start trying to add in loading and error states when we get into these flows - since it's possible the share callback is loading or errors (and our UI won't surface that inforamtion)

Reply:
Yeaaaah.. the annoying thing stopping me from adding all of this to redux is that `useAuth0()` hook, which maintains its own state in a pretty convenient little package. I also think the desire here is to not force everyone to log in before using the UI -- logging in is only required for sharing.

Eventually (soon) I'm going to start focusing on the OpticHub UI, which likely _will_ require logging in to use (except for explicitly public parts, like the spec viewer that'll move over there)